### ISSUE TITLE
CasperJS PhantomJS SSL issues

### ISSUE THREAD

**Issue by vlatkodrljin** [casperjs/casperjs#1241]
When I try run this official CasperJS example:

https://github.com/n1k0/casperjs/blob/master/samples/screenshot.js

with this code addition:

```
 casper.on("resource.error", function(resourceError){
      console.log(JSON.stringify(resourceError));
      console.log('Unable to load resource (#' + resourceError.id + 'URL:' + resourceError.url + ')');
     console.log('Error code: ' + resourceError.errorCode + '. Description: ' + resourceError.errorString);
 });
```

I have following error:

```
Error code: 6. Description: SSL handshake failed
```

and script stop executing.

I use CasperJS 1.1.0 beta3, and PhantomJS 1.9.7

Flags like --ignore-ssl-errors=true and --ssl-protocol=any don't help at all.

Sites without ssl certificate work properly...

Any workaround for this issue?

**Comment by UksusoFF** [comment]
Same issue with 1.1.3 version.

**Comment by jefleponot** [comment]
@UksusoFF   it is a bug of phantomJS 1.9.7

i have got the same error with a pure phantomjs code

```js
var twitterAccount = 'a';
var filename = 'a.jpg';

var webPage = require('webpage');
var page = webPage.create();

page.onResourceError = function(resourceError) {
  console.log('Unable to load resource (#' + resourceError.id + 'URL:' + resourceError.url + ')');
  console.log('Error code: ' + resourceError.errorCode + '. Description: ' + resourceError.errorString);
};

page.open("https://twitter.com/" + twitterAccount, function(status) {
  console.log('Status: ' + status);
  // Do other things here...
});

setTimeout(function(){
phantom.exit(0);
}, 2000);

```../phantomjs-1.9.7-linux-x86_64/bin/phantomjs  --ignore-ssl-errors=false --web-security=false samples/screenshot2.js a a.jpg
Unable to load resource (#1URL:https://twitter.com/a)
Error code: 6. Description: SSL handshake failed
Status: fail
------------
use phantomJS 2.1.0 or SLIMERJS :smile: 

or update certificate authorities .

**Comment by kensoh** [comment]
@vlatkodrljin @UksusoFF closing this issue as the SSL handshake error looks to be fixed with PhantomJS v2.1.1 + CasperJS v1.1.4. But do ping if the issue still persists for you.

**Comment by DonSlockZ** [comment]
@kensoh i'm using PhantomJS v2.1.1 + CasperJS v1.1.4. I've tried to make post requests to my API having self-signed certificate.
If i use only option --ssl-protocol=any  i get "Error code: 6. Description: SSL handshake failed".  Adding --ignore-ssl-errors=true just allows to skip error without solving problem.
Curl and Chrome works fine both https and http. Casperjs can use only http.
SSL handshake error doesn't seem to be fixed.

**Comment by kensoh** [comment]
Hi @DonSlockZ thanks for adding on your findings! Do you mean that using --ignore-ssl-errors=yes works but --ignore-ssl-errors=true does not work? Not sure if this means an opportunity to update CasperJS documentation for PhantomJS page settings.

That setting is not there but I have always used true or false as the flag for this setting.
http://docs.casperjs.org/en/latest/modules/casper.html#pagesettings

**Comment by DonSlockZ** [comment]
@kensoh sorry for disturbing you, it was my fault. Today I noticed I have been using "open" function with the signature of "thenOpen" function. I have been passing "then" callback to "open", so I couldn't get API response status for debug.
Option --ignore-ssl-errors doesn't produce errors, doesn't matter "yes" or "true". Bug was actually fixed in PhantomJS v2.1.1 + CasperJS v1.1.4..

**Comment by kensoh** [comment]
@DonSlockZ oh I get it now, I see.. No probs at all, thanks for taking the time to update here!

It'll help other CasperJS users in future when they search for SSL related issues :smile:
